quinn-proto/src/connection.rs

```diff
@@ -807,14 +790,21 @@ impl Connection {
                             "discarding possible duplicate packet {packet}",
                             packet = number
                         );
+                        if self.check_for_stateless_reset(mux, &packet) {
+                            Err(ConnectionError::StatelessReset)
+                        } else {
+                            return;
+                        }
+                    } else {
                         return;
```

Comment:
This would cause all packets to be ignored. We should instead fall through to the block that calls `handle_connected_inner`.